### ISSUE TITLE
graphql_transport_ws:  Calling `complete` for a single request operation results in a disconnect.

### ISSUE THREAD

**Issue by kristjanvalur** [strawberry-graphql/strawberry#1731]
For the client->server `complete` messate, the protocol spericifation states (boldface mine):
```
Client -> Server indicates that the client has stopped listening and wants to complete the subscription. No further events, relevant to the original subscription, should be sent through. Even if the client **completed** a **single result operation** before it resolved, the result should not be sent through once it does.
```
The problem is, that the message **id** of a single results operation is not placed in `self.subscriptions` and so any attempt at referring to it causes a `KeyError` and a closure.

Now, due to the potential race conditions in the bi-directional Websockets channel, it is impossible to guarantee that a `next` message is not sent after a client sends a `complete` message, so ensureing that a `complete` message from the client actually interrupts a single-request-operation is not of the highest importance.  However, it should not cause a disconnect.

the `complete` message is inherently *racy* because both parties can complete a subscription at any time.  Therefore, both `complete` messages to old subscriptions must be ignored, and a client must ignore `next` or `error` messages to subscriptions it has previously `completed`.

**Comment by kristjanvalur** [comment]
please see https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md which explains how single response requests are just special cases of subscriptions.

**Comment by DoctorJohn** [comment]
This issue has been resolved in #1792. Thanks for the report and the PR :)
